TYPO3.Media/Classes/TYPO3/Media/Domain/Repository/AssetRepository.php

```diff
@@ -278,4 +284,13 @@ public function findAllIterator()
             ->where('a NOT INSTANCE OF TYPO3\Media\Domain\Model\ImageVariant')
             ->getQuery()->iterate();
     }
+
+    /**
+     * @param AssetInterface $object
+     */
+    public function remove($object)
+    {
+        $this->assetService->validateRemoval($object);
+        parent::remove($object);
+    }
```

Comment:
I thought about adding a second argument to the remove() method indeed, but not sure about adding boolean arguments that way... Maybe a forceRemove($object) method...